### PR TITLE
sound: add makefile snippet

### PR DESCRIPTION
**Pull request by alexandermbrown**
Adds Makefile snippet for the sound virtualiser.

### CODE REVIEW

File: sound/components/sound_components.mk [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+#
+# This Makefile snippet builds the sound RX and TX virtualisers
+# it should be included into your project Makefile
+#
+# NOTES:
+#  Generates sound_virt.elf
+#
+
+
+SND_IMAGES := sound_virt.elf
```
Comment (wom-bat) [inline sound/components/sound_components.mk:14]: Is this used anywhere?

File: sound/components/sound_components.mk [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+CFLAGS_sound := -I $(SDDF)/include
```
Comment (wom-bat) [inline sound/components/sound_components.mk:16]: This include should be part of the general SDDF includes already.

Comment (wom-bat) [reply sound/components/sound_components.mk:16]: You can keep CFLAGS_sound around as a way to override flags from the project-level Makefile; but it doesn't have to be anything special.

**Review by wom-bat — CHANGES_REQUESTED** [review]

File: sound/components/sound_components.mk (hunk 1 above)
Comment (alexandermbrown) [reply sound/components/sound_components.mk:14]: Offline: yes, it is used on line 36
https://github.com/au-ts/sddf/pull/154/files#diff-02cc5ea9faacab31e55bdece3fba8034233f5378fe1b6c18f5903483ca734d00R36

File: sound/components/sound_components.mk (hunk 2 above)
Comment (alexandermbrown) [reply sound/components/sound_components.mk]: Done

File: sound/components/sound_components.mk [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+# it should be included into your project Makefile
+#
+# NOTES:
+#  Generates sound_virt.elf
+#
+
+
+SND_IMAGES := sound_virt.elf
+
+CFLAGS_sound :=
```
Comment (wom-bat) [inline sound/components/sound_components.mk:16]: Make it ?= so it can be overridden

Comment (alexandermbrown) [reply sound/components/sound_components.mk]: Done

Comment (alexandermbrown) [reply sound/components/sound_components.mk]: Done

Comment (alexandermbrown) [reply sound/components/sound_components.mk]: Done

**Review by Ivan-Velickovic — APPROVED** [review]

**Review by wom-bat — APPROVED** [review]

**Pull request merged**